### ISSUE TITLE
Renewed issue for (armv7l) Raspberry Pi 4

### ISSUE THREAD

**Issue by sbradnick** [djberg96/sys-filesystem#58]
First off, thank you for this gem :)

I apologize if this is a[nother] report of a known issue, I did read through closed reports where they ended with a 'fixed'-type message.

Here's what I'm seeing:

1. x86_64 odroid h2+ => works fine with version 1.4.2
2. aarch64 rpi4 => works fine with version 1.4.2
3. armv7l rpi4 => does NOT work as expected with version 1.4.2 ; DOES work as expected with version 1.4.1

This is my testing code
```
alarmpi4:/tmp► /bin/cat testing.rb
#!/usr/bin/env ruby

require 'sys-filesystem'
require 'filesize'
include Sys

@filesystem = '/home'
usage_stats = Filesystem.stat(@filesystem)
my_free_blocks = usage_stats.blocks_available * usage_stats.block_size

p Filesize.from(my_free_blocks.to_s).pretty
```

not output I'd expect to see :P

```
alarmpi4:/tmp► ./testing.rb 
"22.90 EiB"
```

Swapping out 1.4.2 for 1.4.1
```
alarmpi4:/tmp► gem uninstall sys-filesystem
Successfully uninstalled sys-filesystem-1.4.2
alarmpi4:/tmp► gem install sys-filesystem -v '1.4.1'
Fetching sys-filesystem-1.4.1.gem
Successfully installed sys-filesystem-1.4.1
Parsing documentation for sys-filesystem-1.4.1
Installing ri documentation for sys-filesystem-1.4.1
Done installing documentation for sys-filesystem after 0 seconds
1 gem installed
```

This is what I'd expect

```alarmpi4:/tmp► ./testing.rb 
"15.35 GiB"
```

Here's /home according to the OS
```
alarmpi4:/tmp► df -hP /home
Filesystem      Size  Used Avail Use% Mounted on
/dev/mmcblk0p2   30G   13G   16G  45% /
```

**Comment by djberg96** [comment]
Maybe it's a 32/64 issue? I'm not familiar with that arch. Can you show me the output of RbConfig::CONFIG?

**Comment by djberg96** [comment]
This was the change between 1.4.1 and 1.4.2:

https://github.com/djberg96/sys-filesystem/pull/57

Based on your paste, it looks like `host_os` is "linux-eabihf" and the `arch` is "armv7l-linux-eabihf". So, it *should* be using the 32 bit code. In unix.rb:

```
elsif RbConfig::CONFIG['host_os'] =~ /linux/i && RbConfig::CONFIG['arch'] =~ /64/
  attach_function(:statvfs, :statvfs64, %i[string pointer], :int)
```
That check should fail since 64 is not in the arch string, causing it use the 32bit statvfs function.

But wait, I see `D_FILE_OFFSET_BITS=64` in cppflags. So, it should be using the 64bit version then, right? Hm, maybe I'll have to check explicitly for that.

**Comment by djberg96** [comment]
Fixed in 1.4.3. Thanks for the report!
